# Joining all datasets with generalized column names

- With hate not included
- What is thumbs up supposed to mean in by_fans page
- Random is included but party will be random
- 0:Unbiased, 1: far-left, 2:centre-left, 3:centre, 4: centre-right, 5:far-right
- Type signifies each page in polly dataset
    - Type 5 is hate and is not included because it does not have party information


In [125]:
import pandas as pd
import glob
import re

In [126]:
def preprocess(row):
    result = re.sub(r"http\S+", "", row['Tweet'])
    result = re.sub(r"RT @\S+", "", result)
    result = re.sub(r"@\S+", "", result)
    result = re.sub(r"\n", "", result)
    return result

In [127]:
data = pd.DataFrame()

In [128]:
page = pd.read_csv('../data/polly/about_party.csv')
page = page.rename(columns={'♡': 'Likes'})
page = page.rename(columns={'About': 'Party'})
page['Type'] = 1
page = page[['Tweet', 'Likes', 'Party', 'Type']]
data = data.append(page, ignore_index=True)

In [129]:
page = pd.read_csv('../data/polly/by_party.csv')
page = page.rename(columns={'♡': 'Likes'})
page = page.rename(columns={'By': 'Party'})
page['Type'] = 2
page = page[['Tweet', 'Likes', 'Party', 'Type']]
data = data.append(page, ignore_index=True)

In [130]:
page = pd.read_csv('../data/polly/by_fans.csv')
page = page.rename(columns={'♡': 'Likes'})
page = page.rename(columns={'👍': 'Party'})
page['Type'] = 3
page = page[['Tweet', 'Likes', 'Party', 'Type']]
data = data.append(page, ignore_index=True)

In [131]:
page = pd.read_csv('../data/polly/with_emoji.csv')
page = page.rename(columns={'♡': 'Likes'})
page = page.rename(columns={'About': 'Party'})
page['Type'] = 4
page = page[['Tweet', 'Likes', 'Party', 'Type']]
data = data.append(page, ignore_index=True)

In [132]:
page = pd.read_csv('../data/polly/with_isa.csv')
page = page.rename(columns={'♡': 'Likes'})
page = page.rename(columns={'About': 'Party'})
page['Type'] = 6
page = page[['Tweet', 'Likes', 'Party', 'Type']]
data = data.append(page, ignore_index=True)

In [133]:
page = pd.read_csv('../data/polly/random.csv')
page = page.rename(columns={'♡': 'Likes'})
# page = page.rename(columns={'👍': 'Party'})
page['Type'] = 7
page['Party'] = 'random'
page = page[['Tweet', 'Likes', 'Party', 'Type']]
data = data.append(page, ignore_index=True)

In [134]:
data['text'] = data.apply(preprocess,axis=1)

In [135]:
data.shape

(106048, 5)

In [136]:
data = data.dropna()

In [137]:
data = data[data['Party'].notnull()]
data = data[data['text'].notnull()]

## Reducing data to just to include two extremes

In [138]:
data = data[data['Party'].isin(['AfD','Die Linke'])]
data = data[data['Party'].notnull()]
data['labels'] = 0
data.loc[data.Party == 'Die Linke', ['labels']] = 0
data.loc[data.Party == 'AfD', ['labels']] = 1
# data = data.rename(columns={'Label':'labels'})

In [139]:
data.to_csv('../data/polly/polly.csv')

In [121]:
data = pd.read_csv('../data/polly/polly.csv')
del data['Unnamed: 0']

In [140]:
data_0 = data[data['Party']=='Die Linke']
data_1 = data[data['Party']=='AfD']
data_1 = data_1.sample(data_0.shape[0])

In [141]:
print(data_0.shape)
print(data_1.shape)

(8185, 6)
(8185, 6)


In [142]:
train= pd.DataFrame()
train = train.append(data_0.sample(frac=0.9))
test = pd.DataFrame()
test= test.append(data_0.drop(train.index))
train = train.append(data_1.sample(frac=0.9))
test= test.append(data_1.drop(train[train['Party']=='AfD'].index))

train.to_csv('../data/polly/polly_train1.csv')
test.to_csv('../data/polly/polly_test1.csv')

In [143]:
print(train[train['Party'] == 'AfD'].shape)
print(train[train['Party'] == 'Die Linke'].shape)

(7366, 6)
(7366, 6)


In [144]:
test.shape

(1638, 6)

In [99]:
train = pd.read_csv('../data/polly/polly_train.csv')
train.shape


(17031, 7)

## Test results on binary data

### For model v2 for unbalanced classes

In [185]:
predicted = pd.read_csv('../data/polly/polly_text_predicted_gelectra_v1.csv')

In [186]:
s = predicted[predicted['labels'] == predicted['prediction']]

In [187]:
(s.shape)[0]/(predicted.shape)[0]

0.6235321747299202

In [ ]:
# train with preprocessing
# learns party specific may not be based on the semantics
# Try zero shot with available models
# Use the model with other parties to see threshold based agreement

## Other parties data

In [17]:
data = data[data['Party'].notnull()]
data = data[data['text'].notnull()]

In [ ]:
del data['labels']

# predicition on different parties

In [18]:
data = data[data['Party'].isin(['FDP','CDU','Die Grünen','SPD','CSU','random'])]
data = data[data['Party'].notnull()]
data['label'] = 0
data['text'] = data.apply(preprocess,axis=1)

In [19]:
data['Party'].unique()

array(['FDP', 'CDU', 'Die Grünen', 'SPD', 'CSU', 'random'], dtype=object)

In [86]:
data.to_csv('../data/polly/other_parties.csv')

In [21]:
predicted = pd.read_csv('../data/polly/other_parties_predicted.csv')

In [22]:
import numpy as np
p = predicted[predicted['Party']=='random']
p_right = p[p['prediction']==1]
print('Percentage right : ', p_right.shape[0]/ p.shape[0])
p_left = p[p['prediction']==0]
print('mean probability for right: ',np.mean(np.array(p_right['probabilities'].tolist())))
print('std: ',np.std(np.array(p_right['probabilities'].tolist())))
print('median: ', np.median(np.array(p_right['probabilities'].tolist())))
print('Mean probability for left: ',np.mean(np.array(p_left['probabilities'].tolist())))

Percentage right :  0.46818067332717656
mean probability for right:  0.9557483476779284
std:  0.09982604482581645
median:  0.9978795051574708
Mean probability for left:  0.03745034753988003


FDP :
Percentage right :  0.5189061886693707
mean probability for right:  0.9631497984544473
std:  0.08861969671054513
median:  0.998441219329834
CDU :
Percentage right :  0.5558645347104976
mean probability for right:  0.970054559927445
std:  0.08430349321980597
median:  0.999220848083496
Mean probability for left: 0.03478617466948551
Die Grünen : 
Percentage right :  0.5072124756335282
mean probability for right:  0.9637299863888367
std:  0.09324209251693516
median:  0.9986621141433716
Mean probability for left:  0.03821723985493388
SPD : 
Percentage right :  0.5511957278848386
mean probability for right:  0.9657167397947902
std:  0.08788509247678603
median:  0.998898983001709
Mean probability for left:  0.03519495763810857
CSU :
Percentage right :  0.6351125938281902
mean probability for right:  0.9697873348165952
std:  0.08227020757262209
median:  0.9990532994270324
Mean probability for left:  0.04239499288147112
random :
Percentage right :  0.46818067332717656
mean probability for right:  0.9557483476779284
std:  0.09982604482581645
median:  0.9978795051574708
Mean probability for left:  0.03745034753988003

In [89]:
p.head()

Unnamed: 0  Unnamed: 0.1  \
0           0             0   
1           1             1   
2           2             2   
3           3             3   
4           4             4   

                                                                                                                               Tweet  \
0                "Die FDP ist noch nicht über den Berg und sollte Demut bewahren", mahnt der Parteigrande Hermann-Otto Solms http://   
1  Solms warnt FDP nach Dreikönigstreffen vor zu viel Hochmut: Der FDP-Bundesschatzmeister Hermann-Otto Solms (FDP)… http:// http://   
2                                                                              Da hat Marco Buschmann vollkommen recht... cv http://   
3                                                                    Live vom Neujahrsempfang der FDP mit Hermann Otto Solms http://   
4                                                                 Leser 2017: Marco Buschmann, Bundesgeschäftsführer der FDP http://   

   Likes Party  Type  \
0      0   FDP     1   
1      0   FDP     1   
2      0   FDP     1   
3      0   FDP     1   
4      0   FDP     1   

                                                                                                                  text  \
0         "Die FDP ist noch nicht über den Berg und sollte Demut bewahren", mahnt der Parteigrande Hermann-Otto Solms    
1  Solms warnt FDP nach Dreikönigstreffen vor zu viel Hochmut: Der FDP-Bundesschatzmeister Hermann-Otto Solms (FDP)…     
2                                                                       Da hat Marco Buschmann vollkommen recht... cv    
3                                                             Live vom Neujahrsempfang der FDP mit Hermann Otto Solms    
4                                                          Leser 2017: Marco Buschmann, Bundesgeschäftsführer der FDP    

   label  prediction  probabilities  
0      0           1       0.991131  
1      0           1       0.999892  
2      0           0       0.000933  
3      0           1       0.968151  
4      0           0       0.000313

# Analysing the failures using v2 so as to identify patterns

In [145]:
test_predicted = pd.read_csv('../data/polly/polly_predicted.csv')

In [146]:
test_predicted.columns

Index(['Unnamed: 0', 'Unnamed: 0.1', 'Tweet', 'Likes', 'Party', 'Type',
       'labels', 'text', 'prediction', 'probabilities'],
      dtype='object')

In [167]:
wrong_predictions = pd.DataFrame()
intermediate = test_predicted[test_predicted['Party']=='AfD']
intermediate = intermediate[intermediate['prediction']==0]
print(intermediate.shape)
wrong_predictions = wrong_predictions.append(intermediate, ignore_index=True)
intermediate = test_predicted[test_predicted['Party']=='Die Linke']
intermediate = intermediate[intermediate['prediction']==1]
print(intermediate.shape)
wrong_predictions = wrong_predictions.append(intermediate, ignore_index=True)

(329, 10)
(359, 10)


In [190]:
from deep_translator import GoogleTranslator
def translate_to_english(row):
    if (not len(row['text'])>5000):
        if(len(row['text'].strip())>0):
            print('---------------------',row['text'])
            english_text = GoogleTranslator(source='de', target='en').translate(row['text'])
    #         print(english_text)
            return english_text
    else:
        return None
wrong_predictions['english_text'] = wrong_predictions.apply(translate_to_english, axis=1)

---------------------   "Solidaritätsdemonstration" nennt der normale Mensch: Unterstützung von Linksterrorismus! #noRAF
---------------------   In einem Salat sind verschiedene Zutaten,warum nicht auch Schnittlauch ?
---------------------  Ich kenne da noch eine Grüne, die hat an einer noch ganz anderen  Demonstration teilgenommen. Ist wohl da ne übliche  Freizeitbeschäftigung.
---------------------   Versuch mal bitte einen Vorteil aufzuzeigen den eine AfD-Wahl hätte.
---------------------    Ich glaube da könnt Ihr lange drauf warten!
---------------------  #ehrlicher #Feind,Lieber als #Falscher #Freund ,ihr wisst was ich mein dann kennt ihr die Farbe vom 🐗und der 🐽#haram außer Himmelblau #BTW17 
---------------------   Schiebt euer illegaler Parteispendeverein, Goal AG, eigentlich Überstunden, um die vielen Grafiken zu produzieren?
---------------------  Grenzen müssen man schützen - ist klar.Sie sollten nüchtern twittern und den Alkohol weglassen.
---------------------   Leider gi

--------------------- Gefängnisstrafe gegen "verbale Angriffe" weil "Argumente" nicht mehr helfen. Ihr habt keine Argumente!! #Nordkorea 
--------------------- Ich wollte doch nur arbeiten! 
--------------------- Gleich geht's los in #Gütersloh!Viele Grüße
--------------------- Start in Rodgau mit  vielen Interessierten im Saal und ein paar Schreihälsen draußen.Wahlkampf macht Spaß.Bundestag-wir kommen! 
--------------------- Ärtzte-ohne-Grenzen-Chef gibt zu: wir retten Menschen, die sich selbst absichtlich in Gefahr bringen. Und die Zukunft ist ihm egal.  
--------------------- Massenmord durch unseren Handelspartner und geschätzten Waffenkäufer Saudi-Arabien. #Aufschrei ? 
---------------------  Solche Phrasen dürfen keine Entschuldigung für die Regierung sein, sich nicht anzustrengen!
--------------------- Überraschung 😀 
---------------------  Beschäftige mich als ehem. Oberst viel mit Sicherheit und Außenpolitik. Aber danke für Ihre Anregung, soziale Themen sind sehr wichtig!
----

--------------------- Wann werden Vorstandsmitglieder von Firmen wie MONSANTO und Bayer vor Gericht gestellt? 
---------------------   1/3 Europa ist ein großer Kultur-Raum, der sich aus vielen kleineren Kultur-Räumen zusammensetzt, die sich aber ähneln.
---------------------  Gibt es einen Link auf diesen Kommentar ? Sieht sehr nach Photoshop bearbeitet aus...
---------------------    Meinungen und Ansichten ändern sich indem man sich damit so intensiv befasst daß sich die Nervenbahnen anders einschleifen. Aber es ist
---------------------    Vielleicht einfach mal sämtliche Zuschüsse streichen. Ärzte innerhalb von Grenzen finde ich zudem weitaus sinnvoller.
---------------------   Das glaubst auch nur du. Ich wohnte neben einer Anlage von 30 Windkaftanlagen.Im Herbst zum Vogelzug Schwäne, Milane und sogar Störche !
---------------------   also in 8-12 Jahren?
--------------------- Fundstück: Wie baut man einen #Staat in eine linke #Diktatur um? Parallelen zu bestehenden Ländern... 
-

--------------------- BILD nimmt Umfrage raus. AFd zu stark? SPD, FDP, CDU, die Linke zu tief?  via 
--------------------- Für gute Arbeit, gutes Geld! In diesem Fall jedoch nicht! 🤔🆘🇩🇪 
--------------------- Norbert Bolz im Gespräch mit Wolfgang Herles  via 
--------------------- Von Müll kommt Müll! 
---------------------   Kein Mensch braucht die Linke....
---------------------    Das Hätschel-Kind der LINKEN
--------------------- Die alten mielke Seilschaften der im-erika? 
---------------------   Jetzt 20 %, bald 33 %
--------------------- Ich will nur noch ausrasten.Jeden Tag... 
--------------------- Unglaublich 
---------------------   Die vergessen dabei immer wieder, dass der Wähler anders tickt. Hier wird es Verlierer geben. Für die Grünen und die FDP wäre es endgültig.
---------------------   Sehr wichtig und richtig
--------------------- Ich will das nicht!
--------------------- Potentieller SeuchenherdBerlin z.B bei HIV/AIDS, Hepatitis B, Legionärskrankheit, Meningokokken

--------------------- so viele wörter sind verbraucht, also sag ich weiter yo, denn es fördert den verkauf.
--------------------- Gehalt für Abgeordnete: Diäten sollen weiter automatisch steigen.  
--------------------- Krisensitzung im Rathaus von Falkenstein (Vogtlandkreis) - Kleinstadt streitet über ihre Nazi-Turnhalle    Wer ständig den Dreck von hinten nach vorne schaufelt,stolpert irgendwann darüber.
---------------------  Sehr geehrter Italo, wie heißt die Quelle (idealerweise in Deutsch)? Ich fand das Schreiben von Hilaire Belloc immer sehr interessant. Danke. Gruß
--------------------- Unsere letzten Regierungen waren einfach  zu blöde zu regieren und liebers wird das Geld der Solidargemeinschaft unnütz ausgegeben , für Menschen die nicht in Deutschland leben.       
---------------------   Gerne. Danke auch. Fürs Mitmachen. Auf weiter guten Zusammenhalt! Herzliche Grüße an Georg, Marc & Kollegen! 👍
--------------------- Vielen herzlichen Dank an all die Mitstreiter, die mir h

--------------------- „Wir brauchen in Deutschland einen sozialen Schub“, sagt  
--------------------- Meine Güte: SPD hat der Union in den letzten Jahren so wenig Contra gegeben. Kritik an #Nahles wird langsam albern.
--------------------- Hier sieht man die Seele der AfD bei der Arbeit, wie es Herr Gauland ungefähr formulierte. Gebürtig und Sozialisiert in Westdeutschland! 
--------------------- Hier noch für meine treuen Follower der Bayernpartei. Die Übernahme Restdeutschlands hat nicht geklappt. PS: fahre gerade nach München
--------------------- Großartig!  
--------------------- Ein furchtbarer Mord, der aufgeklärt werden muss! Und warum haben BMW, BASF & Lufthansa Briefkastenfirmen in #Malta? 
--------------------- #FabioDeMasi im SWR: Besser für #Europa, wenn nicht nur Entscheidungen getroffen werden, die nur #Merkel schmecken 👍  
---------------------  Sie kennen offensichtlich das Jahr 1989 in Deutschland nicht und die Rolle der Bürger in der DDR die friedlich sich ihren Weg

---------------------   Wir sind alle Ochsen vor Gott (Brian Molko)
---------------------   Zumindest, seit Süditalien in die Schulpflicht genommen wurde.
---------------------  Du scheinst den Film aufs Korn zu nehmen ^^
---------------------  Mechaniker... weiß mehr als tausend Wissenschaftler. Genau
---------------------  P.S. Ich hatte auf seinen Tweet geantwortet. Nicht, dass ihr denkt ich wär besonders toll oder  folgt mir. SO cool bin ich nicht😆
--------------------- Auch in der Mediathek... ;-) 
--------------------- Schleschlaaz  Schlechtester Schlag aller Zeiten  #schlefaz
---------------------   Ja modernes Haus und lahmes Internet auf dem Land. Dazu noch dieser schwachsinnige hashtag #fedidwgugl
---------------------  Geb ich gleich weiter. Der funkt noch fleissig. 😃
--------------------- Das selbe Lied Provozieren, dementieren. #noracism #noAfD #Gauland mit Hass gefüllter Rassist ohne Antworten auf Herausforderung - nur Hetze
---------------------   #Kartoffelsuppe nicht v

--------------------- Wenn #AfD'ler über #Flüchtlingspolitik sprechen, isr das immer ein einziges Trauerspiel... 😒☹ #wirwaehlen
---------------------  Oh ja, das sind die #InterRegio... ☹🙄 Fürchterlich... Marode Bahn..
--------------------- Das ist nur noch ein einziges Trauerspiel, was hier mit #Griechenland passiert… ☹
---------------------   Mittendrin im Normalisierungsdiskurs wird dann gefragt, ob man nicht ihre Sorgen um den Volkskörper ernst nehmen sollte, Kompromisse suchen.
---------------------   Wolffsohn hat Kluges zum dt. Antisemitismus gesagt & vertritt dennoch als BW-Uni-Dozent ein geschöntes Staatsräson-Bild vom geläuterten Dtl.
---------------------  Das Verhalten von SA & SS vor der Machtübernahme lässt sich gut mit Pegida, autonomen Kameradschaften und anderen AfD-Konsorten vergleichen.
---------------------   weiß mehr ...
---------------------  Ich wollte ja auch nicht alle wegjagen ;-) Da bin ich dann doch Opportunist genug.
---------------------  Mir fällt es lei

--------------------- Die #Wirtschaftsweisen wollen den #Feierabend abschaffen. Nicht mit uns! Die Antwort auf die… 
--------------------- Eine Krähe hackt einer anderen kein Auge aus. 
--------------------- Es ist an der #Zeit, daß wir uns daran erinnern,daß es auch in #Deutschland einmal eine #Friedensbewegung gegeben hat!Es ist an der Zeit, daß wir den Amerikanern ihre #Atomwaffenmit Dank zurückgeben.Gebt dem #Kaiser was des Kaisers ist!
---------------------   Anschein und Fakten sind halt zweierlei
--------------------- Es gibt jetzt eine Barbie mit Hidschab 
--------------------- Und was taten u. tun unsere Volksvertreter, die diese Bezeichnung mit Sicherheit nicht verdienen? Nicht viel, weil  mit dem Problem müssen sich spätere Regierungen auseinandersetzen.. 
--------------------- Was ein Quatsch!Skandal in französischer TV-Show: Lagerfeld nennt Flüchtlinge "schlimmste Feinde" der Juden  via 
---------------------    Das muss man nicht, ganz im Gegenteil!Wer friedlichen Demonst

---------------------  👍✊!?Wie die Folgen offener Grenzen finanziert werden sollen,dazu sagt das Programm nichts.
--------------------- Bis 59 Jahre wär  locker drin. Darüber kippt es unter die 5%. (Mehr Ost wär gut gewesen ;) Bei unter 25jährigen 9%. 👍 


In [191]:
wrong_predictions.to_csv('../results/wrong_predictions_v2_to_analyse.csv')

In [192]:
correct_predictions = pd.DataFrame()
intermediate = test_predicted[test_predicted['Party']=='AfD']
intermediate = intermediate[intermediate['prediction']==1]
print(intermediate.shape)
correct_predictions = correct_predictions.append(intermediate, ignore_index=True)
intermediate = test_predicted[test_predicted['Party']=='Die Linke']
intermediate = intermediate[intermediate['prediction']==0]
print(intermediate.shape)
correct_predictions = correct_predictions.append(intermediate, ignore_index=True)

(2326, 10)
(1244, 10)


In [193]:
correct_predictions['english_text'] = correct_predictions.apply(translate_to_english, axis=1)
correct_predictions.to_csv('../results/correct_predictions_v2_to_analyse.csv')

--------------------- AfD-Sprecher Martin Renner kandiert auf der NRW-Liste für den Bundestag auf Platz 1. Pretzell setzt auf Export aus Hamburg Kay Gottschalk.
--------------------- Gaaaanz knapp bei der Stichwahl für Platz 1 der Liste der #AfD #NRW für den #Bundestag: Martin Renner siegt soeben vor Kay Gottschalk.
--------------------- Gratulation meinem Hamburger Parteikollegen Kay Gottschalk zu Listenplatz 4 der AfD NRW zur Bundestagswahl. #afd  #btw2017 #afdessen
---------------------   Die Grünen sind Linksextremisten.
---------------------  Wer grün wählt, unterstützt den Linksextremismus, nicht die Natur, siehe EEG, Windräder, Klimalüge.
---------------------   Die Direktübertragung sollte künftig organisiert werden...
---------------------   Schon mal drüber nachgedacht WER das bezahlt? - DU zahlst das (solltest Du Steuern zahlen) und ich, etc. DAS findest Du in Ordnung ?
--------------------- Wieder eine tolle Wahlkampfveranstaltung der  mit  und  bei der  #TrauDich #AfD 
---

---------------------    Meines Wissens nach ist kostenlose Werbung eine geldwerte Zuwendung & damit eine Parteispende, aber das klären Gerichte #AfDSwissConnection
--------------------- NEW VIDEO by Jörg Meuthen Wir sind im Sorge um unsere Heimat, Vaterland und Zukunft unserer...
---------------------    Sie ist ein großer Mensch
---------------------   Verzeihen Sie den Fehler, ich meine selbstverständlich Herrn #Arppe
---------------------    Ein EU Einwanderungsgesetz, ja, unbedingt. Hat aber mit Asylrecht kaum etwas zu tun. Und Asylbewerber einfach zurückweisen ist unmenschlich.
---------------------  Das eine: Zum anderen würde ich mal empfehlen zu go-ogeln nach: dr alice weidel goldman sachs; Finanzkartell NO WAY!
--------------------- NEW VIDEO by Wahlkampfveranstaltung der #AfD Südthüringen mit Jörg Meuthen, Stephan Brandner
---------------------    Und was soll jetzt genau damit sein? Richtig "Rechts" sowas oder?
---------------------   Falls Sie Ihren 'illegal', 'OECD' Quats

---------------------  Wech mit dem Halunken
---------------------  Wer Pol Pot zur Machtübernahme gratuliert, der steht mit Heinrich Himmler moralisch auf einem Podest.
---------------------   Linksversifft und dumm ... und von Demokratie keinen blassen Schimmer ...
---------------------   Natürlich. In den Augen der #afd ist die NSU Patrioten und die Opfer nur morgenländische Subjekte. #noafd 
---------------------   Sie beten aber wohl nicht den christlichen Gott an bei so unchristlichen Kommentaren oder? Bitten Sie Gott lieber um Benehmen und Respekt!
---------------------   Sie zitieren wirklich einen Blog der die Menschenjagt durch Linksradikale fördert?
--------------------- Jörg Meuthen zerfetzt die Bundesregierung in der Luft - Abensberg 04.09.17  via 
--------------------- NEW VIDEO by AFD-Television:#AfD   Prof  Dr  Jörg Meuthen   Audio Kolumne vom 30 09 2017 Eine historische Woche
---------------------    Die braucht niemand.Sie meinen also die "fremden Völker", die nach D.

---------------------  Komisch im Duden(Onlinefassung) wird national als Synonym für nationalistisch verwendet. D.h. dann wohl aus #AfD-Sicht.. #Lügenduden 🤣
--------------------- Als SLDP Politikerin rufe Ich nachWiderstand&Resistance gegen die AfD ! AfD ist Staatsfeind nr1in BRD! Gemeinsam gegen alle AfD Staatsfeinde
--------------------- LKW tötet. Messer tötet. Waffe tötet. Nein, der MENSCH der diese bedient, tötet. #Schrottpresse #noIslam #AfD 
--------------------- #NSAfD Anspruch und Wirklichkeit ... - #NSAfD Anspruch und Wirklichkeit ... AfD-Chef Alexander Gauland hatte Ay... 
---------------------   ja, und da wissen sie leider auch ganz genau, sie sind nämlich nicht so doof wie alexander gauland aussieht.
--------------------- Dr. Alexander #Gauland, Fraktionsvorsitzender der #AfD im #Bundestag, lehnt die Forderungen der Präsidentin des Umweltbundesamtes, Maria Krautzberger (SPD), unter anderem nach einer allgemeinen Maut für alle PKW auf sämtlichen Straßen, entschieden ab.➡️

---------------------  Deutsche haben Millionen Juden getötet, weil sie glaubten, das Judentum als DAS Problem schlechthin identifiziert zu haben. Heute gibt es Deutsche, die glauben, den Islam als DAS Problem schlechthin identifiziert zu haben. Die Hetze der #AfD ist Goebbels' Arschgeburt.
---------------------  doch:wenn der bürger diese aufgabe übernimmt..so wie früher..
---------------------  Ein offen zur Schau gestelltes Christentum auch, Kreuzzug mit umgekehrtem Vorzeichen
---------------------  Es gibt 80 Millionen Deutsche, davon ca. 1 Million Gutmenschen. Diese eine Million verteten die Deutschen Medien. Ein Präsident von Amerika wird auf das übelste von der Politik und den Medien verteufelt. Frau Merkel wird von den selben vergöttert. Dieses Spiel verstehe ich nicht.
---------------------  Kein Experte in Politik zitiert keinen Experten in Politik! #Lagerfeld
---------------------  Toll, wie sich Herr Professor jetzt für jüdische Mitbürger einsetzt! Schaffen Sie es auch, Ihr

---------------------    🌐Sehr cooles Werbefilmchen !Die Zeit als Neuling ist vorbei! Die AfD hat Volsparteistatus erreicht!☝️🌐
---------------------  Laßt die #Spiele beginnen, ciao Mutti. Don Giovanni
---------------------   ...endlich mal seine widerliche Fresse halten!
---------------------   Jedenfalls finde ich nicht, dass die  die pseudokünstlerische Provokation dieses  tolerieren sollte. Irgendwo ist Schluss! 
---------------------   ...als auch der fabulierende, weltfremde  -Professor Pfeiffer!!
---------------------   Ein CDU-Politiker wird angegriffen wegen seiner Politik.Das war Grund des Talks. Tweet von maischberger erzeugt Eindruck, dass gerade ein messerangriff auf einen afd-Politiker stattgefunden habe.Dabei hat afd den Eindruck erzeugt, Flüchtlinge seien Grund für leistungskürzungen.
--------------------- Alice Weidel kritisiert die Medien: Eine ehrliche Berichterstattung gibt es nicht mehr von  via  
--------------------- Wer würde sich nicht gerne von Alice Weidel a

--------------------- Interessanter Artikel :-)  Die Blaue Partei: Frauke Petry und Marcus Pretzell haben noch lange nicht genug  
--------------------- Neben Frauke Petry steht mein Politik&Wirtschaft-Lehrer auf der Liste von Leuten, die ich gerne überfahren würde.
--------------------- Was macht eigentlich diese selbsternannte "#Realpolitikerin" Frauke #Petry so?Außer in zwei Parlamenten fette Diäten einstreichen, wohl nicht viel, oder?  #Die_Blauen
---------------------  Ich habe Sie beim Parteitag gesehen und musste feststellen, dass Sie mir noch immer sympathisch sind. Weiterhin gutes Gelingen und einen schönen 2. Advent für Sie und Ihre Familie. 🕯️🕯️
---------------------    O Gott! Haben die es auch getan? 😂😂😂
---------------------   Qualifiziertes Personal wäre durchaus vorhanden, nur ob Höcke-Pazderki so einen Spagat hinbekommen würden? Zmnd. fraglich, aber die Partei ist jung und hat Potenzial nach oben. Da ist noch Luft!
---------------------  Frauke Petry macht das mit 5 Ki

--------------------- Anja Kling ist aber attraktiver als Frauke Petry. #tatort
--------------------- Diese Rechtspartei kommt uns bekannt vor:  Regisseur Niki Stein hat einen „Tatort“ gedreht, der wie ein Schlüsselfilm über die AfD wirkt, die Hauptfigur erinnert stark an Frauke Petry: „Dunkle Zeit“ (von Heike Hupertz) 
--------------------- Hat Frauke Petry das Drehbuch zu diesem #Tatort geschrieben oder nur die Buchvorlage?
---------------------   Ich wünsche dir das auch, Georg.
---------------------   Danke! dto.
---------------------  Danke gleichfalls! Und viel Erfolg!  🤗
---------------------  Danke, ihnen und ihrer Familie ebenso
---------------------  Und vertragt euch mal da in eurer Partei!!!!Ich habe euch nicht gewählt damit ihr genauso einen Müll macht wie die anderen Raffkes des Altparteienkartells!
---------------------   Das stimmpt!  😊
---------------------     sollten diese Schriftstücke keine Fälschung sein ist das ein Skandal, wenn in Deutschland von Amtswegen Straf

---------------------  Wenn es um die AfD geht, werden mittlerweile sämtliche Register gezogen und das Ganze erfolgt noch nicht mal mehr subtil. Man muss sich ja nur mal anschauen, welchen Posten die Tochter von Herrn Schäuble und Ehefrau von Herrn Strobl innehat!
---------------------     Haben Sie sich in dem Film wiedererkannt ?
---------------------    Frauen werden von ihren Glaubensbrüder nicht nur angemacht sondern übelst belästigt.Um Konflikte zu vermeiden empfehle ich Ihnen in  ihre Heimat zurückzukehren, Sie  werden dort sehnsüchtig erwartet. Wir werden mit unseren Assis selbst fertig, die nicht im Ansatz so schlimm sind
---------------------    Keine Kolonialmacht.
---------------------   Ich kritisiere die #AfD da, wo es angebracht ist. Basta.
---------------------   Im ZDF gibt's nicht wenige Sokos über 'böse Rechte'.
---------------------   Interessiert nur keine 🐖
---------------------   Wie eben Die Tagesschau/-Themen und Heutenachrichten/-Journal. Nur noch Mainstream-B

--------------------- #AfD-Programm zur #Btw17, Teil 9:🔹Verhältnis zu 🇷🇺 entspannen🔹Sanktionspolitik beenden🔹Zusammenarbeit vertiefen➡️ 
--------------------- Libyer. #Messerjihad 
--------------------- #TrauDichDeutschland!Nach dem 24.9. wird es keine weitere Gelegenheit mehr geben, diesen Wahnsinn zu stoppen.#AfD.
--------------------- 402,9 Millionstel, also ein Spurenelement, aber verdummt die Bürger nur weiter ... 
--------------------- Da will man mal in Ruhe im Zug arbeiten- und wen treffe ich da? Mannmannmannmannmann. 
--------------------- Mit ihr wird es eine E-Quote nicht geben. #Maut 
--------------------- Wer Merkel wählt, entscheidet sich für jemanden, der den Rechtsstaat dem eigenen politischen Vorteil unterordnet!➡️ 
--------------------- Viele Fragen, viele Antworten. Tolle Möglichkeit, dem Wähler den klaren Kurs der #AfD vorzustellen.#TrauDichDeutschlad #AfD #FragSelbst *
---------------------   Die spannende Frage ist doch nicht, wie unsere Gesetze lauten. Die Frage 

--------------------- Warum fragt #wahlomat nicht nach Islam,4. Griechenlandpaket,Familiennachzug,GEZ,Bargeldabschaffung,aber nach Bafög, Nutztieren u Impfpficht?
--------------------- Unfassbarer Tiefpunkt der politischen Diskussionskultur in 🇩🇪:#SPD-#Stegner vergleicht die #AfD mit Parasiten!➡️ 
--------------------- #AliceWeidelLive zum #TVDuell:Merkels Flüchtlingspolitik zerstört den Sozialstaat 🇩🇪:Die #AfD wird dafür sorgen, dass das ein Ende hat!
--------------------- Merkel u Schulz wollen nicht dt.Grenze schließen,sondern die des Tschad u beide wollen Familiennachzug. Dolles #tvduell  Danke für die Frage
--------------------- Mehrwertsteuer senken! #AfD wählen! 
--------------------- ... und antworten mit Thesen aus dem #AfD-Programm. #TVDuell Wegen erwiesener Unehrlichkeit hilft dies CDUCSUSPDGRÜNEFDP aber auch nichts. 
--------------------- Bitte packen Sie mit an und unterstützen uns dabei, zweistellig nach der #Btw17 in den Bundestag einzuziehen!#TrauDichDeutschland: #AfD
-

---------------------  Och nööö.
--------------------- ❗️+++ACHTUNG+++ ❗️Account  ist ein F͟A͟K͟E͟!Die abstrusen Aussagen dort haben mit der #AfD NICHTS zu tun!
--------------------- Auftritt in Heidenheim muss krankheitsbedingt leider ausfallen.#TrauDichDeutschland #AfD #Btw17Veranstaltung ⛔:   *
--------------------- Jetzt  : "Ich bin  gläubiger Christ. Nur mit der Amtskirche habe ich Probleme." 🙏 
--------------------- Hohle Phrasen, leere Versprechen und Beschwichtigungen:🇩🇪 braucht endlich eine e͟c͟h͟t͟e Opposition!#Btw17 #TrauDichDeutschland #AfD
--------------------- Hat nix mit dem Islam zu tun. 
--------------------- Nichts als Schande! #GEZ muss weg. 
--------------------- Da hat die FAZ ganz recht. Die AfD könnte an der 5%-Hürde scheitern. Ist doch so. Die CDU aber auch. 😂 
--------------------- Welt erfindet Kehrtwende, d. es nicht gibt. Lesen hilft! Springe weiter nicht über Stöckchen dieser Schmutzkampagne! 
--------------------- Kein Gedenken und besser keine Fragen:Terr

--------------------- Eines ist sicher: Es kommen ungemütliche Zeiten auf diese kollektiven Rechtsbrecher durch die  zu!➡️ 
--------------------- Sollten für "schwer Traumatisierte " nicht andere Maßnahmen notwendig sein? 
---------------------  warum Tschüss? 😀
--------------------- 200.000 Flüchtlinge (ohne Familiennachzug!) ohne Anspruch mehr p.a., denn 🇩🇪 ist umgeben von sicheren Drittstaaten. 
--------------------- Den Medien stünde es gut zu Gesicht, derlei Gefahren für die normale Bevölkerung nicht kleinzureden!#AfD #LtwNds➡️ 
--------------------- Die einen warnen, mahnen, fordern. Die anderen verschleiern, täuschen, versagen. Und urplötzlich- ist die Lage dann "außer Kontrolle". 
--------------------- Bundesbank-Bilanz kaputt. 
--------------------- Selten wurde Steuergeld in 🇩🇪 so gut, nachhaltig und zum Wohle des deutschen Volkes investiert! ☑️😉#AfD #LtwNds
--------------------- . zur "Wirtschaftskraft" Berlins, die nach einer Analyse das BIP je Einwohner drückt!#AfD➡️ 
----

--------------------- Die  wird mit einstweiliger Verfügung gg. Verleihung & Anzeige gg. Gruppe wegen Verleumdung vorgehen!➡️ 
--------------------- Der Twitter-Account des 🇺🇸-Präsidenten sei "versehentlich aufgrund eines menschlichen Fehlers für 1️⃣1️⃣ Minuten deaktiviert gewesen"! 🤣 
--------------------- Guten Morgen 🇩🇪!Frau  spricht wahre Worte aus & bestätigt die unzähligen Warnungen der #AfD hierzu!➡️ 
--------------------- Es ist ein Skandal. Unsere Polizei wird vollständig korrumpiert. 
--------------------- Ein Lamento der Journalisten vom #Staatsfunk. #GEZabschaffen 
--------------------- Persönlicher Hilferuf von  gegen Stasi-Tanten umd Consorten. Ich bin dabei! 
--------------------- »CDU will offensichtlich Frage nach ihrer Identität gar nicht aufkommen lassen, damit Koalition zustande kommt.« 🤔 
---------------------    Ist das Ihr Ernst?
--------------------- Leider "vergisst"  im guten Artikel zu erwähnen, dass  all das seit Bestehen anprangert! 
--------------------- P

--------------------- MdB  spricht für die #AfD-Fraktion zum Bundeswehreinsatz gegen die Terrororganisation IS #COUNTERDAESH: 
---------------------   Vor was sind die Terroristen, die „Flüchtlinge“ waren, geflohen?
--------------------- CDU will auch keine FREIWILLIGE Rückkehr syrischer Flüchtlinge nach Syrien. Das sei menschenverachtend.  #GutZuWissen #AfDimBundestag
--------------------- MdB Rüdiger #Lucassen (❞Die Grünen & Herr Lindner wollen den Eid der #Bundeswehr auf den ganzen Planeten ausdehnen!❝#MINUSMA
--------------------- Dr.  zu SPD-Einwanderungsgesetz: ❞Erhöhung der Geburtenrate & aktivierende Familienpolitik statt eigenes Volk auszutauschen!❝ 
--------------------- #AfD-Bundessprecher Prof.  im  (incl. Audio):❝Die Politik, die Frau Merkel gemacht hat über die letzten Jahre hinweg, ist in vielen Politikbereichen eine Politik multipler Rechtsbrüche.❞
---------------------    Falls es nicht direkt abgespielt wird, auf den Link unten klicken und dann unten rechts auf den Pf

--------------------- Bitte hören Sie sich hier mein Interview mit dem SWR an: #AfD 
--------------------- Peter Felser #AfD: Die kaputtgesparte, marode und ausgelaugte Bundeswehr muss endlich gestärkt werden! #Daesh #IS 
--------------------- Auf die Zwischenfrage unseres Abgeordneten  antwortete SPD-Mann Fechner im #Bundestag tatsächlich:❞Die Zeiten der #Sippenhaft sind l͟e͟i͟d͟e͟r͟ vorbei.❝Ein Rechtsverständnis, das wahrlich für sich selbst spricht.#AfD #AfDimBundestag➡️ 
--------------------- Trotz erheblicher Bedenken des Bundeskriminalamts entschloss sich der  damalige Bundesfinanzminister Wolfgang Schäuble dazu, die Untersuchung von Geldwäsche auf den Zoll zu verlegen.➡️  
--------------------- Die einzigen Armbändchen, die wir hier angebracht sehen, sind Handschellen an den Händen derer, die unsere Frauen und Töchter bedrängen, berauben oder sexuell übergriffig werden - und in deren Kultur eine Frau keinerlei Wert hat.#Silvester #Reker #AfD➡️ 
--------------------- Aus der Schw

--------------------- Livestream aus Frankreich: Auto rast in Gruppe von Soldaten – Sechs Verletzte, davon zwei Schwerverletzte 
--------------------- Sauberkeitskampagne der Stadt stößt auf Kritik Duisburg: Jammer-Türken beklagen „rassistische“ Mülleimer 
--------------------- USA: Ist das DNC eine kriminelle Bande?
--------------------- Was Präsident Macron vorbereitet, von Thierry Meyssan Ja Franzosen falsch gewählt bei uns gibt es auch Vollpfosten
--------------------- Webseiten, die Adblocker blockieren sind nach meiner Meinung Mafiöser Natur zumindest ist ihre Kundenfreundlichkeit am Ar....
---------------------    Rechts aber nicht zwangsläufig radikal. Radikal ist, wer Gewalt anwendet.
---------------------   Das ist keine Degradierung, das ist unsere Natur.
---------------------   das nach meiner Meinung nach aufgebauscht wird. 😉
---------------------   Der Memet ist ein feiner anständiger Mensch solche Menschen sind Goldstücke Herr NiewirstduKanzler.
---------------------   D

---------------------   Oh du Schreck die Elite des ätzenden grünen Schleims😁
---------------------   Welcher Punkt des AFD Wahlprogramms ist bitte rassistisch? Und warum darf sich die AFD nicht zu den Terroranschlägen äußern?
---------------------   Müssen Sie nicht verstehen. Die Opfer verstehen das auch nicht. Die nächsten Opfer werden auch wieder überrascht sein. Alles gut soweit...
---------------------  Antifa von der Straße prügeln, inhaftieren und zu Schadenersatz zwingen! Würde die Arbeit der Polizei erheblich erleichtern!
---------------------  Eine weitere Frage: Wieviele #Gruene haben ihre Haustür stets geöffnet und lassen jeden rein so lange dieser will? #GrueneVersenken
---------------------  Gold ist eben so wertvoll das es nicht arbeiten braucht...
---------------------  keiner von denen hat einen zu Hause und bürgt persönl für ihn. Wir haben sie alle da und bürgen alle. Mit € und Leib und Leben. #Sozialismus
---------------------  Und noch amüsanter  wie die Trolle all

---------------------  Falsch! Wir brauchen eine Politik die Familienplanung fördert und die Familie als Keimzelle der Gesellschaft stützt#TrauDichDeutschland
---------------------   Aber ihr von den Grünen habt von nichts eine Ahnung
---------------------  inklusive der kompletten  versteht von allem gar nichts. Deswegen #GrueneVersenken 
---------------------  Art. 16a Abs 2 GG was gibt es da noch zu entscheiden?
---------------------     Und danach stellt sich eine brunzdumme Hetzerschlampe wie Sarah hin und macht "Rechte" für die Gewalt verantwortlich,
---------------------    Ich kenne viele ü 60,die wissen wohin die Reise nach der BTW geht.Sie wählen blau,aber die vielen Dumpfbacken ,Rautefans, unverständlich!
---------------------   Bitte, begründen Sie Ihre Unterstellung, dass wir unglücklich sind. Weil wir #AfD wählen?
---------------------   War mir fast klar,das aus dieser Ecke sowas kommt.Das sind die Totengräber von🇩🇪
---------------------  Dann lieber so weitermachen wie 

---------------------  Alles nur Propaganda aus Angst vor breiten Zustimmung aus dem Volk.
---------------------  Cool 😎 muss ein Russe sein ein bio deutscher wäre schon Tod 💀
---------------------  Damit auch morgen der NSA mithören kann 👏👏
---------------------  Danke. Habe herzlich gelacht! Wahlkampf ohne Plakate!Und Martin gewöhnt sich schon mal daran, dass er bald neben der AFD sitzt. 😂😂
---------------------  Das hat Polen an die Sowjetunion verloren, was hat Deutschland damit zu tun? Die SU ist 1945 genauso in Polen einmarschiert.
---------------------  Das übliche Geschwurnel
---------------------  Denk dir doch ne neue videokategorie aus 🤗
---------------------  Der Widerstand gegen was bitte???Nazis^^Wer ist dat bitte?Sind es die 80 90 jährigen die die Zeit erlebt und ja auch gelebt haben?
---------------------  Die faschistische SAntifa ganz in Tradition a la 1933... Andersdenkende bedrohen, verprügeln, einschüchtern... Adolf wäre stolz auf euch!
---------------------  FDP i

---------------------    Im Moment ist es die Vorraussetzung überhaupt mal gegengeprüft zu haben.
---------------------    Skandal, aber Gabriel hat seinen Kumpels im Kindergarten die Mettwurst vom Brot weggefressen! 🤔🍗🥓🌭🍩
---------------------   Das der einzahlende alleine damit auskommen muss, dass das nicht geht und nicht reicht, wissen wir doch alle!
---------------------   Der schaut gerne öfters mal tief ins Glas,u.hat sich heute morgen vielleicht  vor seiner Rede schon einen oder zwei genehmigt.
---------------------   Diesen Artikel muss man 2x lesen! 🆘🇩🇪 
---------------------   Du setzt "Es gibt Klimawandel seit es Klima gibt" (#AfD Programm) mit Klimaleugnen gleich? Da haste aber schon mal besser argumentiert.
---------------------   Erklären Sie mal, warum ich 10 offene Hilfsarbeiter-Stellen, bei 2500 gemeldeten arbeitssuchenden Flüchtlingen, nicht besetzen kann???
---------------------   Hehehee, dass ist richtig!
---------------------   Mach mal halblang. Sind ja wohl all

--------------------- Wie erbärmlich #FDP! Haut ja einen von d Schuhen sowas dämliches! Wenn echt, möchte ich den sehen dr ein paar Kreuzern so hinterher jagt! 
--------------------- Wieder eine Contra #AfD Sendung mit Anti-Demokraten besetzt #maischberger hau ab!
--------------------- Wir sollten alarmiert sein, von denen die z.Z. im Parlament sitzen!😡🆘🇩🇪 
--------------------- Wusste ich doch, #Fakenews  Zitat, Anstellung einer Asylbewerberin erfolgte n͟i͟c͟h͟t͟!
--------------------- 👍🤘 stimmt 
--------------------- "Als ich erstmals gewählt wurde, gab es noch kein Internet." #RedenWieMerkel #Klartext #BTW17
---------------------   Tja, dann müssen Sie die AfD wählen, sonst wird das nichts.
---------------------  ...und den #Bundestag aufblähen! Gute Chance jetzt auf 800; und dann? Pöstchen sichern = 1. Ziel der #Politiker!
---------------------   "VATER STAAT" ein sicheres Zeichen, dass man es mit einem Vollidioten 1. Klasse zu tun hat.
---------------------   etwas gegeben, aber n

---------------------  Amis sind bekloppt!
---------------------  Andreas Wild wurde ausgeschlossen.Was er sagte,gilt nur für ihn und nicht für die #AfD
---------------------  Im eurem Video geht es also um einen Tweet? Von wem denn?Hier ein paar Infos dazu.
--------------------- an der vernichtung ist merkel & soros, beide juden beteiliegt! aburteilen und entsorgen ! seit schlau wählt blau- afd ! 
--------------------- Der obersozi mit seinem oberlehrerhaften gesülze ist sowas von ätzend. 
--------------------- die steinbach hat recht ! meine hochachtung ! 
--------------------- Geschlossene veranstaltung? Hier gibt es keine antifa. Irgendwie doch kommisch. Steckt die CDU oder etwa Im-Erika selbst dahinter? 
--------------------- Gutmenschen: Hatte sich gefragt. Sie wählt trotzdem die CDU! 
--------------------- In den BRD-Amtsstufen hing bisher immer #Gauck. Jetzt #Steinmeier? Ist Personenkult! Weg damit! Deutsche Fahne hin!
--------------------- Leipzig Bürger Versammlung Wg moschee

---------------------  Das Büchermädchen hat noch mehr Probleme … 
---------------------  Den Respekt, den Sie verdienen, schenkt der Staat an andere weg, die nichts Konstruktives leisten werden, wenn sie nicht sogar zerstören
---------------------  Die Afd ermöglicht uns allen hoffentlich einen Aufbrauch, damit sich die Heimat nicht weiter ins Negative verändert! Paradoxe Zeiten
---------------------  einmal Krimineller immer Krimineller:D
---------------------  Wir sind prinzipiell für mehr Demokratie und Volkssouveränität. Sogar dann, wenn einmal „Linke“ davon profitieren würden. #Katalonien 
--------------------- Da soll noch 1 sagen #wählen+Volksverrätern die Meinung geigen bringe nichts"Deutschland,von der #Willkommenskultur zur Abschreckung"#LTWNDS 
--------------------- Dieser Quatsch. Ich bin mit Computern aufgewachsen und nutze sie täglich und wie kann die AfD, dann so multimedial dauerpräsent sein? 
--------------------- Es ist eigentlich ehrlich, wenn Macron seine Verachtun

---------------------  na,wenn dies nicht Programm ist, ‚unbequemen Flüchtlinge‘ los zu werden. Damit wäre das aktive Gestalten einer Parallelwelt in🇩🇪
--------------------- Deutschland wird sich verändern. #87Prozent freuen sich drauf. #EU #Volksfeste #Städte Innengrenzen statt Außengrenzen. 
--------------------- Diktatoren treten nicht zurück, sie müssen gestürzt werden! 
--------------------- Gute Nacht Deutschland 🇩🇪 
--------------------- INTERVIEWS, RUNDGANG, BUCHVORSTELLUNGVideo: Björn #Höcke auf der #Buchmesse #fbm17
--------------------- "Der Vorfall mit dem Filmteam ereignete sich bereits am 17. Juni." Dichtgehalten bis nach der Wahl👍 #Staatsfernsehen 
--------------------- 37 jähriger richtet Blutbad an bei seiner Arbeit. 3 Kollegen tot, 2 in kritischem Zustand 
---------------------   Du trittst für #Toleranz und #Meinungsfreiheit ein und willst anderen verbieten, wenn sie die #WAHRHEIT sagen. 😂JA, „in #Gelsenkirchen sind ganze Stadtteilen gekippt, nur durch den Zuzug von 

---------------------  Der #Bundestag war im letzten Jahr schon Comedypreis-Sieger. #Gabriel #Roth #KGE #Kauder #Merkel usw. haben den Preis sich redlich verdient
---------------------  Der Grüne Quotentürke bricht eine Lanze für seinen Sultan und so einer soll Außenminister der BRD werden
---------------------  Er wurde NACH der Schließung gestellt - als Reaktion auf #G20. Es geht um die linksextremistischen Bewohner, die unter Milieuschutz fallen.
---------------------  Typisch Linksfaschisten
---------------------  Wer nicht selber vorsorgt handelt grob fährlissg bei einer "Regierung", die ihren Job nicht macht.
---------------------  Ralf Stegner macht mal wieder den Goebbels.
--------------------- Dafür aber das ABC der #Waffensachkunde 😂 
--------------------- Ist sie hoffentlich nicht mehr lange 
--------------------- Sexismus bei #AktenzeichenXY -> Sagt der doch "Guten Morgen, schöne Frau". Das geht doch gar nicht!
--------------------- So viel Vernunft und Mut zur Wahrheit ver

---------------------    Shhhh Holger.. Sie labern bullshit!
---------------------   👍👍👍😂😂😂.... Wenn ich Zipfelmann lese, denke ich auch eher an Pimmelmann.... Genau wie auf dem Foto 👍👍👍😂😂😂
---------------------  Ausländer?😂👍 
---------------------  Uschi repariere erstmal deine Waffen und dann reden wir  . Pfoten weg vom demokratischen und stolzen Polen!!!!
--------------------- Zum Glück gibt's keine Steuer auf's öffentlich dumm und hässlich sein. Sonst könnte die komplette grüne Partei nie das Haus verlassen. 
---------------------  Dem Täter war der Besitz seiner Tatwaffe lt. Waffengesetz verboten. Setzen, sechs und ab auf den stillen Stuhl.
---------------------   FDP, schon immer Vollpfosten !
---------------------   Diese Idee war sowieso totaler Schwachsinn, kann man nun auch sehen !
---------------------  das Gesocks soll den Mund halten die haben nichts zu melden die sind nur Gast in unserem Land
---------------------  Wird auch höchste Zeit, Waschlappen !!
------------------

--------------------- Polizei findet Sprengstoff am Weihnachtsmarkt 
--------------------- FAILED STATE NRW MADE BY CDU 
---------------------    Und die gesamte derzeitige Situation im Nahen Osten hat die Obama Administration verzapft. Der Nobelpreisträger. Ahnungslos und zu zögerlich.....
---------------------  Weitsicht mit Steuergeldern.... Was für ein Held..oder doch ein Narr?
---------------------  Ganz einfach: Sie ist links und keine Dame.
---------------------  Das ist das typische AfD bashing, das umso stärker betrieben wird umso stärker die AfD wird und umso mehr auf Neuwahlen zugeht!
--------------------- Mannheimer OB spricht von „Staatsversagen“ - Minderjährige Flüchtlinge sorgen für Ärger     Seit zwei Jahren ist dieses Staatsversagen bekannt u.die Regierung Merkel ignoriert es einfach.
--------------------- Sone #InternetKrawallmacher wie #DushanWegner folloow ich liebend gern! 😃 
---------------------    Ich weiss 👌👌👌
--------------------- Die ist schlimmer.Sie vernich

--------------------- Mannheim: Schläger-Teenies greifen Polizisten auf dem Weihnachtsmarkt anÜber meinen Google-Feed geteilt 
--------------------- SPITZEL DER DDR 
--------------------- Widerliches Dreckschwein dieser Neger !!!! 
---------------------    Nöööö!!!! Aber ein Datenvolumen, daß "zu Fuß geht".... Bin halt zu geizig🙅🙇
---------------------   Artikel gelesen! Man muss ja erst mal sehen, welche Folgen....hätte mich auch nicht gewundert, falls "Corona-Träger" 'ne Lichterkette für den Traumatisierten organisiert hätten!? 
---------------------  ICH BIN INTEGRATIONSWILLIG.... 
---------------------  Wir auch....😴 
--------------------- Also - dann mal "auf die Socken " machen!!! Sonst bleibt für euch der Platz unterm Baum leer!!!!😭👎🎄▶🚫
--------------------- KAPITULATION vom Feinsten...Fortsetzung der damaligen Äusserung bezgl. der Grenzöffnung /Ungarn..."Was sollten wir denn tun?"Abgelehnte Asylbewerber: De Maizière verteidigt Prämie für freiwillige Ausreisen - DeutschlandÜber 

--------------------- Muslimische #Bürgerwehr auch hier in #Münster. 👎  #Islam #Scharia #TrauDichDeutschland: #AfD  
---------------------   Mal ehrlich, dieser Religionsduselei ist doch schuld an den weltweiten Konflikten! Die sollte man allesamt verbieten!😡
---------------------   das ist doch selbstverständlich! 👍
---------------------   Wünsche ❤ Frauke⚘ und ihren Baby eine SICHERE und GUTE ZUKUNFTund der AfD 60 %👍 Wünsche mir und für uns alle:Frauke als Bundeskanzlerin❣
---------------------  Was auf Landesebene bewiesen, soll nun auch bundesweit wirken. Sinnloses populistische Bombardierungen nur um Diäten abzukassieren. AfD 👎
--------------------- Es gibt gewisse Ausdrücke,die ich einfach nicht mehr hören kann.  👎 #afd #hamburg 
--------------------- In der Flüchtlingskrise wurden warnenden #AfD-Anhängern Aluhüte angedichtet.Heutzutage tragen Polizisten stich- und kugelsichere Helme. 😡😡 
---------------------  ALEXANDER GRAU: zu Populismus 👍👍👍 
---------------------  Das sind di

---------------------    Also etwas wundervolles wie Geburten mit Krieg in einem Wort zu verbinden, ich schreibe lieber nicht was ich davon halte...👎
---------------------  Diese Sendung ist völlig daneben.Dämliche Fragen und Antworten. Nur gehetze auf Frau Weidel und die Afd! 👎
---------------------  ja, vor allem wie nah #fdp bei #afd liegt!#WahlOMat 😡 
---------------------   Libanesen garantiert sauer auf deutsche #Polizei ! 🤔Haben doch eigenen "Friedensrichter"?! 🤔#Parallelgesellschaft #Paralleljustiz 😡 #AfD
---------------------   Sat 1 und Strunz bescheissen das Publikum beim Viererduell am 30.8.2017 Krankenschwester Fake😡 
---------------------  Wahnsinn! Was ein Skandal! Man stelle sich vor, ein AfD Mitglied hätte sich dort so dargestellt😡
---------------------  Wid die SPD eigentlich vom Verfassungsschutz beobachtet? 🤔Nein, gehört zum Regime 😡
--------------------- Ich wusste ich kenne die Dame. Nun weiss ich woher ... #FakeNews bei #sat1 😡 sehr neutral diese Medien ...   
--

---------------------  Der soll ja auch mit der AfD zusammenarbeiten, nicht mit seinen Hirngespinsten! So ein Dödel, echt mal.👎
---------------------  Selbstherrlich und vollkommen realitätsfern. Leider wird so die SPD noch weit tiefer fallen. Macht nur weiter die AfD stark 👎👎👎
--------------------- Eine Hirnzelle weniger und du wärst ne Pflanze. #Bashing gegen die #AfD #Demokratie in #Deutschland geht anders 👎👎👎 
--------------------- #Maischberger Endlich!Nicht das 1.Wort zur Begrüßung: AfD! Wir sind auf dem Weg zur Ignoranz gegenüber den Rechten!! Zu spät, Wahl vorbei 😡
---------------------  😂😂"eintreten" & die Ankündigung anderen eins auf  "Fresse" zu geben. 😡😡und ich bin keine "AfD" Wählerin
---------------------   Nö, wollte damit sagen, dass viele gayleute #AfD gewählt haben und das mit gutem Grund, nix Alibi! 👍👍👍💣🔫🔪😈😈😈
---------------------   😉👍trotzdem - positiv lächeln ist gesund
--------------------- Erst Mandate abstauben und dann gehen? Danke  das sie  platzmachen👍 #AFD #

---------------------   Sehr gut, dass Sie so etwas tweeten.  Von den Systemparteien habe ich sowas noch nie gelesen 👍👍
---------------------   Na wenn das so ist, dann dürften Sie ja keine Schwierigkeiten aus der Situation herauszukommen. Ich wünsche Ihnen viel Erfolg. 👍
---------------------    Seltsame Aussage, wo doch die #AfD bei anderen Aussagen sehr wohl applaudiert hat 👎👎
---------------------   Möchte  uns so glauben machen.Nun wird wohl neben standardmäßigem "kostet Arbeitsplätze" auch mit "stärkt AfD" argumentiert.👎
--------------------- Da fehlen einen die Worte. Einen von der  das Gesicht eintreten 😱 Tweet muss man natürlich nicht zensieren 😡 
--------------------- Insektensterben -ein ökologisches Armageddon❗️Regierung & #Grüne denken an KlimaPOLITIK😡 als an UmweltSCHUTZ!#AfD 
---------------------    Es läuft alles im Sinne von Blut-Merkel, die den NATO-Auftrag Umvolkung ausführen will❗👍😎
---------------------    Es wäre aber eine andere Strafe möglich gewesen ! Oder nic

--------------------- Es nimmt seinen Lauf 👍😂#afd #noafd #hoolkopf 
--------------------- Ob #NetzDG, #Diätenerhöhung, linker #Terrorismus oder offene Grenzen ...die #AfD rockt den Bundestag und spricht die Probleme an. Gegen den Widerstand aller Altparteien. 👍
--------------------- Phantastisch! 👏👌Danke, Frau  👍 
--------------------- Rede von  zeigt wieder sehr deutlich wie unangenehm den Kartellparteien die deutlich abweichende Meinung der #AfD ist.  für mehr Meinungsvielfalt und mehr Wahrheit im #Bundestag. 👍💙🇩🇪
--------------------- 👍nur 20? Ein Blick in den Bundestag, von ganz Links bis kurz vor die AFD Bänke und vorne komplett, da finden sich doch sicher noch massig solche Kandidaten, oder? 😄 
---------------------  Frage ..Frau Weidel??Wo ist man in Deutschland 🇩🇪 noch SICHER??🤔🤔 Germany is a danger Country 🇩🇪🙃👎👎
---------------------    Hat sowas von .... 1939. 👎
---------------------    Igitt, was ist das denn für ein IQ-gebremster Zeitgenosse #twitternwieposener  👎👎👎
-------

---------------------  Ja schade 😕 Ich mag das nicht wenn Politik/ Politiker mit so (AfD) wischi waschi Methoden die Bevölkerung für dumm verkauft. Aber die 7 - 12% stimmen da ja noch zu ... 👎
--------------------- Zum Kotzen das Weib! 👎👎👎👎🤢🤢🤢💩💩💩 #zdf #ard #cdu #merkel #afd #spd #bild #breitscheidplatz #terroristen 
--------------------- Die Anzahl der islamistischen Gefährder in Deutschland🇩🇪 ist dramatisch angestiegen. CDU-CSU😨, SPD😡, GRÜNE🤮+LINKE😠, es ist das Ergebnis Ihrer Flüchtlingspolitik😵 Kommt es durch Sie zu Anschlägen, sind es dann wieder "Einzelfälle".😫 #Islam #AfD  
--------------------- nur die AFD kann diese politischen nieten im bundestag blossstelleen! jahrzehnte konnten diese "kümmerlinge" in der regierung ihre suppe kochen da keiner reinschauen konnte! wo sind die modernen kreuzritter deutschlands,der spuck muss ein ende haben!🚔🇩🇪😡🤢😠 
---------------------   Ja..dumm wie kleine konditionierte mäuse die nicht hinterfragen was die afd denen verschweigt😀👍 
-------------

--------------------- Alice Weidel ist ein Bot
--------------------- Diese  ist eine unverschämte Person. Wie kann sie mich nur so von der Seite anreden. Weiß sie nicht wer ich bin?  #ZDF Chefredakteur Frey: #Slomka will "Auseinandersetzung in der Sache", moderierte "fair + gelassen". Hat er die Sendung gesehen?🤭#weidel
---------------------    Alice Weidel ist eine Hurentochter.
--------------------- Alice #Weidel ist eine von uns. Sie verschafft uns ein Gehör ob der Ungerechtigkeiten in unserer schönen Heimat. wählen
--------------------- #AfD Trulla Alice Weidel ist eine staatsfeindliche Rassistin?Hätte ja wirklich niemand drauf kommen können.Oh.
--------------------- Was?  ist eine Nazi-Uschi?!WER HÄTTE DAS GEDACHT!?Kids, ALLE AfD'ler sind Nazis. Sie sind Nazis in Businesskostümchen.
---------------------  ist ein politisches Prachtexemplar der #AfD#HolDirDeinLandzurück#MutzurWahrheit#TrauDichDeutschland#BTW17#Bundestagswahl1:37Dr. Alice WeidelVerified accountUnangenehme Fragen an 

--------------------- VenezuelaEva Golinger ist wie Sahra Wagenknecht mit scharfer Intelligenz 😎   #kenfm #grüne #spd
--------------------- Wahlkampfthema soziale Gerechtigkeit - u.a Sahra Wagenknecht bei Anne Will | 27.08.2017 
--------------------- #Stuttgart Sahra Wagenknecht, Dietmar Bartsch, Bernd Riexinger und Johanna Tiarks - Musik: Msoke und Moderation: Cuno Hägele 
--------------------- Gregor Gysi, #DIELINKE: »Die Riester- #Rente ist ein Hohn«  via 
--------------------- Schau dir "Gregor Gysi sagte die Wahrheit über Bundeskanzlerin Angela Merkel" auf YouTube an... 
--------------------- Jaja Katja Kipping, bezahlbarer Wohnraum, wieder eine nicht funktionierende Mietpreisbremse? #Wahl2017
--------------------- Katja Kipping ist als Einzige überzeugend. Bei Lindner und Weidel würgt es mich!
--------------------- Katja Kipping trägt auf den Wahlplakaten eine Art FDJ-Bluse – dort wäre sie wohl in einer fortbestehenden DDR auch hauptamtlich gelandet.
--------------------- Katja K

---------------------   Gregor Gysi - alle Deutschen sind Nazis.
--------------------- Gregor Gysi: "Frieden und Sicherheit" in Europa nur mit Russland an Bord möglich  
--------------------- Ich mag das   Gregor Gysi über Martin Schulz: "Man kann nicht über Nacht einfach
--------------------- Ich mag das   Katja Kipping nimmt die Hetzer der AfD auseinander.
--------------------- phoenix: Linken-Chefin Katja Kipping: "Fortschrittliche Mitte Links-Lager erste...  via 
---------------------   Habt ihr Plätze für zwei attraktive Ostberliner aus dem Dr.-Gregor-Gysi-Wahlkreis?
--------------------- #content Katja Kipping sagt was Sache ist! Elefantenrunde 24.09.2017  via 
--------------------- Katja Kipping hat sehr interessante Wangenknochen.
--------------------- Katja Kipping wäre als Margot Honecker eine oscarreife Besetzung. #lanz
--------------------- Frauke Petry verliebt sich gerade in Gregor Gysi #maischberger
--------------------- Gregor Gysi hat auch im Hinblick auf Söder die ric

--------------------- Gregor Gysi über Ausstellung "Russische Revolution" - "Stalin ist einfach ein Verbrecher" 
--------------------- Katja Kipping sollte man zum Spargelstechen nach Brandenburg abkommandieren, damit sie mal weiß,... 
--------------------- Katja Kipping von  wählt also "keinen Nazi" zum Vizepräsidenten des Bundestages!🤔 , Anwalt fragen, ob dies justiziabel ist? 
--------------------- Ich habe ein Video zu einer  hinzugefügt:  Linke Lästereien: Bernd Riexinger schickt subtile
---------------------   Von Gregor Gysi&Israel.Und den Moslems,den besten Freunden der Juden.😂Die bezahlen auch Steuern.Sie kämpfen auch fleißig gg Antisemitismus
--------------------- ESM - der Weg in die Diktatur Gregor Gysi erklärt die wahren Folgen... 
--------------------- Ich mag das   Gregor Gysi: Frau Merkel, ihre Aussagen hier haben nichts mit der Realität zu
---------------------  Es ist schade,für die guten Leute wie Sahra Wagenknecht u Lafontaine,die reiben sich auf zwischen den Idiote

--------------------- Gregor Gysi, Präsident der Europäischen Linken: "Europa befindet sich in einer strukturellen Krise." Namhafte Redner auf dem Abschlussplenum des größten Einkäufergipfels Europas, der am Freitag in Berlin beendet wurde.  
--------------------- #dielinke #hessen der Landesparteitag beginnt: alle Menschen habe gleiche Rechte! Kein Wackeln in der Flüchtlingsfragen! Und wir sind froh, Bernd Riexinger als Parteivorsitzenden zu haben :) 
--------------------- #dielinke Parteitag in Hessen: große Zustimmung (Bernd Riexinger): für eine lebendige Mitgliederpartei, die schnelle stark ist auch außerparlamentarisch und in sozialen Brennpunkten präsent. Nicht Wahlverein oder "Zujubelparte" ;)
--------------------- ++ Zynischer geht's kaum ++Sahra Wagenknecht rechnet mit linker Flüchtlingspolitik ab ..#dieLinke
--------------------- Sahra #Wagenknecht: „Wir müssen uns allerdings immer wieder bewusst machen: Die schlimmsten #Steueroasen sind nicht auf den Bahamas, sondern mitten 

--------------------- GOOD NEWS (y) Gute Initiative , liebe  Katja KippingLängerfristig 'sorgenfrei durch den ganzen Monat' würde ein... 
---------------------    Wäre es nicht besser, die Politikerin aus Thüringen Sahra Wagenknecht zu wählen?
--------------------- DAS! Sahra Wagenknecht in der NDR-Talkshow - Wagenknecht - 
--------------------- Wahnsinn, dass Sahra #Wagenknecht nicht viel früher dieses „Interview“ abgebrochen hat! #ZDF #Satire 
--------------------- Lieblingsmenschen für immer Gregor Gysi und Mario Basler
--------------------- Akif Pirinçci an die Nachfolgerin von Mördern, die Bolschewistin Katja Kipping 
--------------------- Linken-Katja Kipping hat die AfD für die Messerattacke auf den Bürgermeister von Altena, Andreas Hollstein (CDU), mitverantwortlich gemacht. 
--------------------- Also fast kann sie einem Leid tun. Fast. Das besondere Interview mit Sahra #Wagenknecht  #Rütten #MannSieber
--------------------- Liebe Sahra Wagenknecht, genauso ist es. Dieser Mann

--------------------- Gregor Gysi sagt 1998 (!) die Zukunft des Euro exakt voraus! -  
--------------------- Oscar Lafontaine als saarländischer Kriminalkommissar, der bis zum Punkt, an dem eine Riesenspinne Sahra Wagenknecht privat, aber auch politisch, ersetzt hat, nicht wahrhaben will, dass die Spinnen ein Problem sind. Danach dann Brucewillisisierung von Oscar Lafontaine.
--------------------- Sahra Wagenknecht demaskiert CDU und SPD  via 
--------------------- Sahra Wagenknecht, Die Bürger wollen Sie nicht mehr Frau Merkel!  via 
--------------------- Wo Sahra Wagenknecht recht hat, da hat sie recht! 🇩🇪 
--------------------- Dr. Gregor Gysi (Mitglied des Deutschen Bundestages) referiert auf dem 18. #BHB - Kongress in #köln über „Sicherheit-Gerechtigkeit-Freiheit“ #shkjournal 
--------------------- Ausgerechnet Katja #Kipping macht auf Nächstenliebe!  via  
--------------------- Bin gerade in einer Telefonkonferenz #dieLinke und Katja Kipping - sehr interessant!
------------------

--------------------- Bodo Ramelow hat als einziges Argument die angebliche Beleidigung, dass  er mit dem Vornamen angeredet wurde. Verständlich zwar bei dem Vornamen,  aber irgendwie auch armselig. 
---------------------    Was macht Gregor Gysi da?
--------------------- Heute: Abend der Neumitglieder von  Special Guest:  Vorsitzende von  Ab 19:00 Uhr #Diskussion über zukünftige #Herausforderungen der #LINKEN. Ab 21:00 Uhr laden wir zur Disco. Mehr Infos hier:  
--------------------- Ein Deutschland ohne Niedlichlöhne und Altersarmut 05.09.2017 Sahra Wagenknecht – Bananenrepublik 
--------------------- Die Parteivorsitzenden Katja Kipping und Bernd Riexinger nehmen Partei für Diether Dehm und weisen den Artikel... 
---------------------   Weil ebenjene Antisemiten im Antiimp-Spektrum kräftig Wählerstimmen fangen. Wäre Sahra Wagenknecht nicht *das* mediale Zugpferd der Linken und die einzig relevante Talkshowpersönlichkeit, wär die auch schon lang weg vom Fenster (will ich jedenfalls h

---------------------  Frohe Weihnachten ! Nie zuvor waren wir so gut informiert übers ReGIERungsUnwesen . Trotzdem bleibt Die LINKE unter 10 % . Schade . 
---------------------  In dem Moment wo er aufschlägt ist die linke Hand neben dem Ball. Erinnert an Dez Bryant im Playoff
---------------------  Nachdem die linke Gesinnungsjournalistin und Marxistin Sonia Mikich Sebastian Kurz wie eine Aktivistin der Antifa beleidigt hat, gibts sie auch noch völlig unumwunden zu, dass sie für das Merkel-Regime Journalistin spielt.
--------------------- Der Berliner CDU-Politiker Kai Wegner kommentierte auf Twitter: „Unfassbar. Die linke Szene macht nun Hetzjagd... 
--------------------- Die Polizei fahndet mit Fotos und Videos nach G20-Gewalttätern. Die linke Szene reagiert promt und droht den Beamten indirekt. #G20 #G20HAM17  
--------------------- Große Freude herrscht bei der Fraktion DIE #LINKE & #PIRATEN über das #Glyphosat-Verbot für #Dortmund  
--------------------- Puh. Ein Glück, dass ich

--------------------- Wir wollen, dass der Mindestlohn in der Pflege 14,50€ beträgt. Eine Pflegerin muss mindestens 3.000€ brutto im Monat verdienen. 
--------------------- Der #Migrationsgipfel von #Paris war in Wahrheit nur ein Abschottungsgipfel. #Fluchtursachen bekämpfen geht anders. 
--------------------- Russland-Sanktionen sind wirkungslos und kontraproduktiv, Frau #Merkel. Sicherheit kann nicht auf Konfrontation aufbauen. #BTW17
--------------------- Aktuelle Armutsquoten zeigen: Bundesregierung und Andrea Nahles haben bei der Armutsbekämpfung kläglich versagt. 
--------------------- Solche Armutsquoten sind rote Karte für #Merkel.  steht für 1050€-Mindestsicherung. #duverdienstmehr #btw17 
--------------------- Armutsgefährdungsquote ist auf dem Höchststand. 20,7% ist die Quote bei Kindern. So wollen wir #Armut bekämpfen:  
--------------------- Ich danke ausdrücklich dem scheidenden Innenminister Holger Poppenhäger für seine Arbeit und zolle Ihm meinen vollen Respekt!
-------

--------------------- "Schluss mit den #Waffenexporten an #Saudi-Arabien, an die #Türkei. Das muss endlich aufhören!"  mit  #BTW17 
--------------------- Mein Interview im SRF: «Wir wollen nur regieren, wenn es Partner gibt, die den Sozialstaat wiederherstellen.» #BTW17 
--------------------- Jetzt live aus #Bitterfeld   #linke #BTW17
---------------------   Programmieren und IT-Kenntnisse an Schulen und für jedes Kind ein mobiles Endgerät als Bildungsausstattung. #digitalwahl #digitallinks
---------------------   Indem wir in unsere öffentlichen Hochschulen investieren und die Forschung stärken. #digitallinks #BTW17
--------------------- Das Geld dahin schaffen, wo es gebraucht wird! Managergehälter begrenzen, Reichtum #Umverteilen und investieren in Bildung und Gesundheit. 
--------------------- Jetzt im Livestream  mit Sören Pellmann:  #Linketruck #BTW17 
--------------------- Welche Gründe gibt es für junge Menschen DIE LINKE zu wählen? live:  #BTW17 #Leipzig
--------------------- 

---------------------   Und Katalonien wurde von Franko verfolgt und unterdrückt! Ja, es braucht ein neues Verfassungsrecht.Deshalb werbe ich für einen Föderalstaat
--------------------- Eine soziale "Allianz für Ostdeutschland" ist Vorraussetzung, um Spaltungen & Ungleichheiten zu überwinden. #TDE2017
--------------------- Für soziale #Gerechtigkeit und eine Politik ohne Arroganz. Dafür am 15.10. in #Niedersachsen mit der #Zweitstimme DIE LINKE wählen! #LTWNDS 
--------------------- Rechtliche Wehr bei Arbeitslosenversicherung lohnt: 44% der Widersprüche ganz oder tlw. erfolgreich, 34%der Klagen. 
--------------------- Liebe Autofahrer, ein Fahrradstreifen ist ein Fahrradstreifen und KEINE 2. Parkspur!
---------------------   Auch die! Selbstverständlich! Gute und Ideenreiche Unternehmer, davon können wir gar nicht genug bekommen.
---------------------   #Heimat ist ein Gefühl  #Heimat ist eine Sehnsucht Warum sollte man diese Gefühle freiwillig dem Nationalismus überlassen?
---------

--------------------- Das gilt auch im Umgang mit dem Autor Wolfgang Schorlau. Ich habe bei der Entstehung des Romanes aktiv mitgewirkt,es ist&bleibt ein Roman! 
--------------------- Auch #280Zeichen werden die Welt nur verschieden interpretieren, es kommt darauf an, sie zu verändern.#Duverdienstmehr
--------------------- Am Montag durfte ich die neue ICE-Hochgeschwindigkeitsstrecke durch #Thüringen testen. Ein kleiner Bericht ist im Tagebuch nachzulesen:  #ZukunftThüringen
--------------------- Jetzt live im Interview mit Claudia Langer zum #Generationenmanifest auf Facebook: Generationenmanifest.de 
--------------------- Die Banken lassen die Finger nicht vom Glücksspiel - trotz „rien ne va plus“ in der Finanzkrise. Fast jeden Tag kommen neue Skandale an Licht, die zeigen, dass weder Banken strenger reguliert und kontrolliert noch Steueroasen trockengelegt worden sind. #ParadisePapers 
--------------------- Also manchmal flattern  Einladungen in meinem Briefkasten wie die zur Preisv

---------------------   Dresdner Pädagoge Gustav Schanze 1906:Hausaufgaben in Volksschulen sind vom unterrichtlichen Standpunkte aus  als entbehrlich anzusehen.1958 Erziehungswissenschaftler Bernhard Wittmann "Hausaufgaben besitzen keinen materialen Bildungswert, bewirken keinen Zuwachs an Kenntnissen"
--------------------- "Atmend" scheint ein Zauberwort zu sein, wenn die #Gruenen bei was mitmachen. Erst die atmende Fabrik von #Hartz, jetzt atmende #Obergrenze, #atmenderrahmen #AtmenderDeckel. #JamaikaSondierung
--------------------- Schnell noch ein Interview zu unserer Forderung Kinderrechte ins Grundgesetz gegeben. Nun heißt es Luftballons aufblasen für den Kindergeburtstag. Die Party beginnt gleich. #Weltkindertag 
--------------------- Unsere Forderung zum Tag der Kinderrechte: Kinderrechte ins Grundgesetz! Eine Mehrheit dafür wäre möglich. 
--------------------- Die Sondierung haben gezeigt, dass FDP und CSU so weit rechts stehen, dass es selbst mit  und  keinen Kompromiss gab. 

---------------------  Wenn Leichtgewichte Theater veranstalten. #keinarschinderhose 
--------------------- . berichtet von den Scheinprozessen in der #Türkei. Wir fordern die türkische Regierung auf, alle inhaftierten #HDP-Abgeordneten und Journalisten freizulassen. #FreeDeniz #Demirtas #Yueksekdag #Ankara 
--------------------- "Wir möchten  beim Wort nehmen." Eine #SPD, die Massenentlassungen hinnehme, "unterschreibt ein AfD-Förderungsprogramm für den Osten", fügte sie mit Blick auf den von #Siemens geplanten Stellenabbau hinzu.  #spdbpt17 #schulz  
--------------------- Schon bei den ersten Sondierungen in der nächsten Woche könnte #Schulz die Weichen für einen sozialpolitischen Kurs stellen. Eine #SPD, die die Massenentlassungen unter Merkel hinnimmt, unterschreibt ein #AfD-Förderungsprogramm für den Osten. #Sachsen #Thueringen #duverdienstmehr 
--------------------- Herzliche Glückwünsche an  zur Wiederwahl als stellv. Vorsitzender  Hessen - Thüringen und an den neuen Vorsitzende

--------------------- Das lehrreiche an #Twesten ist doch, wie nah sich #Grüne und #CDU inzwischen sind. Von wegen alternativ.
--------------------- Ohne #68er keine Freiheit heute - Es gab, gibt und wird nie eine gute alte Zeit geben- wer #68er verurteilt, hat Geschichte nicht verstanden
--------------------- Schöner Beitrag von  Wir sagen aber: Parteispenden von Konzernen ganz verbieten! #Lobbyismus 
--------------------- 😇😀 Putin ist erdverbunden und hat gesunden humanistischen Menschenverstand. 
--------------------- #FDP-Vorsitzender  fordert Annäherung an #Moskau -  tobt 
--------------------- #Landschaftspark #Duisburg. Immer einen Ausflug wert. #NRW #Ruhrgebiet @ Landschaftspark… 
---------------------  Das macht den #Diesel auch nicht besser. Ihr Argument: "Dort wo schon viel Dreck ist, können ein bisschen #Diesel nicht stören" ist zynisch.
---------------------  ...als Kanonenfutter für das US-Imperium andienen. Kriegstreiber, egal woher, sind abzulehnen.
--------------------

---------------------  Die Bayern wollen ein neues Ministerium, eines für Plagiate und offizielles Versagen! #ZuGuttenberg  ist optimale Besetzung dafür
--------------------- Die Linken leben gerne und machen dementsprechend positive Politik, die alle Menschen mit in die Teilhabe am Leben einbezieht! 
---------------------  Wie denn sonst? Es gibt so viele zu klärende Fragen, klar hat sie dafür keine Antworten! Also werden nur Fragen zugelassen, wo sie glänzt
---------------------    Da haben sie auch keine Hebamme! Aber Herr Spahn braucht dringend einen Hebammer, der ihm einiges 'ganz spontan' erklären muss!
---------------------    Seit 2005 war die #SPD acht Jahre lang an der Bundesregierung beteiligt. Nicht gut für alle. Soziale Ungerechtigkeit steigt dank SPD.
---------------------  Das bGE würde aus unserem verkrusteten Dienstbotendenken eine soziale Mitmenschlichkeit ermöglichen.
---------------------  Die sog. Moderation des #TVDuell  ist so unbefriedigend wie ihr überflüssiger

--------------------- Die vielen Fälle, von denen wir nie erfahren, weil nicht angezeigt/ eingestellt/Opfer mit Überleben beschäftigt... mir ist grad schlecht
--------------------- einfach lesen... 
--------------------- Ich beschreibe jetzt eine Szene, für die ich seinerzeit eigens eine Besichtigung im Flughafen Tempelhof mitgemacht habe im März 2016. #Schreibnacht
--------------------- Ich habe keinen Bock mehr, keinem gut genug zu sein.
--------------------- Trolle, die für das Trollen bezahlt werden, nennt man Astroturfer.#infotweet
---------------------    Meiner ist nur halb besetzt, aber wenn ich mehr als 30 fahre, beschwert sich mein Kleiner.
---------------------   Ein paar wenige haben S-Pedelecs mit Nummernschild, sind aber eher die Ausnahme.
---------------------  Aber die AfD will doch nur das wiederbringen, was sich im Nachhinein als schlecht herausgestellt hat. Schlecht gewählt, würde ich behaupten.
---------------------  Dann schau dir 1fach den Rest der Serie an und er

--------------------- #Abu_DhabiAuch bei der Zusammenarbeit in Kunstprojektensollte man die Fragen von #Demokratie und #Freiheit nicht aus dem Blick verlieren.#FreeDenis#FreeMesale#FreeThemAll, (auch schweizer #Journalisten in #Wüstenstaaten!)
--------------------- Eine Anschauempfehlung für alle, die die letzten Wochen die Auseinandersetzungen in der Linken irgendwie verfolgt haben. Ich finde, damit kann es wieder vorwärts gehen. 
--------------------- Ich würde mich gerne beruflich neu orientieren und mein (mehr als nur) Hobby Politik zu meinem Lebensinhalt machen. Falls jemand in der Partei DIE LINKE oder bei der RLS Interesse an einem Mitarbeiter hat, kann sich gerne bei mir melden. 
---------------------   Oh, den Grünen hab ich wohl übersehen. (ruhig konkreter antworten, was du meinst) wär auch zu schön gewesen. Ändert nichts an meiner Grundaussage.
---------------------  Den Fachkräftemangel habe ich als frischer Akademiker am eigenen Leib zu spüren bekommen. 7 Monate Jobsuche.


---------------------   Laschet wünscht sich mehr Sachlichkeit… WTF! Ist das nicht der Mann, der gesagt hat: "Ich schaffe Euer Sozialticket ab. Bätschi! Bätschi! Bätsch💥" … 👿👿👿 🤕 "Oke. If faffe Euer Fosialticket doch nif abb…" 😔
--------------------- Welches Projekt aus ihrem Wahlprogramm sollte die SPD in einer möglichen neuen Großen Koalition auf jeden Fall umsetzen?  #CiveyFrage via  
---------------------    Es braucht: Gewissenhafte Diagnostik und bei ADHS dann multimodale Therapie, bei der Ritalin oft ein wichtiger Baustein ist.
--------------------- Die USA will wieder eine bemannte #Mondmission. Sie sollten den gesamten Trump-Clan mitnehmen, am besten auch Kim Jong-un, Erdogan so wie die Reste des Kapitalismus  - und dann dort vergessen!  
--------------------- In zwei Tagen kann ich 17.000 Euro für den iMac Pro lassen, wenn ich möchte… :D
--------------------- Wie ich Magneten aus einer Hülle ausbaute. 
--------------------- "Die sich vor der Sonne fürchten müssen"Update zu „L

---------------------  Geht schon seit Jahren so, und die Politik greift nicht durch. Man hat das Gefühl, die linke Szene genießt Narrenfreiheit. 😡
---------------------   Die Linke hat sich schon eine ganze Weile unwählbar gemacht wie alle anderen Altparteien auch!Aber das jetzt ist doch ein Geschmäckle!😖👎👎
---------------------    Als ich heute morgen auf der A66,Richtung ffm gefahren. Wurde ich von Rollsplitt beschossen. Die hatten die linke Spur geflutet 😡
---------------------  Die Linke ... danach hab ich nicht mehr weitergelesen 👍
---------------------  Wir sprechen immer über rechte Gewalt, die linke Gewalt darf auch nicht geduldet werden! Dank an jeden einzelnen Polizisten 👍
---------------------  Die Linke kann natürlich direkt raus aus dem Bundestag. Dann haben die gewalttätigen Idioten auch kein Sprachrohr mehr. ❤️
---------------------   Stellungnahme von Die #Linke zu Hamburg. #G20 👎 
--------------------- 100%👍 „Es ist zum Kotzen, dass die Linke nicht bereit &fähig ist, 

--------------------- Gregor Gysi ist ein guter Mensch #lastRT
--------------------- Da haben wir es: Gregor Gysi ist ein Nazi... 😆N24Verified accountLinken-Abgeordnete erklärt SED zur rechten Partei 
--------------------- Herr Gregor Gysi ist ein sehr guter Redner und Politiker. Doch DIE LINKE besteht nicht nur aus Herrn Gysi. One Man Show is out.  Ich seh und lese viel  dieser Tage. Es macht immer wieder Laune. Eine positive Kraft! Bin froh, dass alle an einem Strang ziehen.  …


## Analysis of words based on the PMI score

In [210]:
from collections import Counter
import math
from sklearn.feature_extraction.text import TfidfVectorizer
from matplotlib import font_manager
from sklearn.feature_extraction.text import CountVectorizer
from collections import OrderedDict

In [217]:
def get_top_n_features_based_pmi(n, left_tweets, right_tweets, left=True):
    n_left = len(left_tweets)
    n_right = len(right_tweets)
    left_document = ' '.join(left_tweets)
    right_document = ' '.join(right_tweets)
    document = ' '.join([left_document,right_document])
    cv = CountVectorizer(stop_words='english')
    cv_fit = cv.fit_transform([left_document, right_document])
    features = cv.get_feature_names()
    
    counts = cv_fit.toarray()
    print(len(features), counts.shape)
    pmi = []
    for index, f in enumerate(features):
        N = (n_left + n_right)
        p_x = (counts[0][index]+counts[1][index])/N
        p_x_left = counts[0][index]/N
        p_x_right = counts[1][index]/N
        p_left = n_left/N
        p_right = n_right/N
        
        try:
            pmi_x_left = math.log2(p_x_left/(p_x * p_left))
        except ZeroDivisionError:
            pmi_x_left = 0
        except Exception as e:
            pmi_x_left = 0
        try:
            pmi_x_right = math.log2(p_x_right/(p_x * p_right))
        except ZeroDivisionError:
            pmi_x_right = 0
        except Exception as e:
            pmi_x_right = 0
        
        combined_score = (pmi_x_left - pmi_x_right)
        if left:
            combined_score = (pmi_x_left - pmi_x_right) * p_x_left
        else:
            combined_score = (pmi_x_right - pmi_x_left) * p_x_right
            
        pmi.append({'feature':f,
                    'pmi_hate': pmi_x_left,
                    'pmi_neutral': pmi_x_right,
                    'score': combined_score})
    
    pmi_df = pd.DataFrame(pmi)
    pmi_df = pmi_df.sort_values(by=['score'], ascending=False)
    print('left: ', left, pmi_df.tail(5))
    top_n = pmi_df.head(n)
    print('top_n: ', top_n)
    return top_n

In [218]:
pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)
pd.set_option('display.width', 1000)
print(wrong_predictions.shape)
wrong_predictions = wrong_predictions.dropna()
print(wrong_predictions.shape)
top_n_left = get_top_n_features_based_pmi(100,
                             wrong_predictions[wrong_predictions['labels']==0]['english_text'].tolist(),
                             wrong_predictions[wrong_predictions['labels']==1]['english_text'].tolist(),
                            left=True)
top_n_right = get_top_n_features_based_pmi(100,
                             wrong_predictions[wrong_predictions['labels']==0]['english_text'].tolist(),
                             wrong_predictions[wrong_predictions['labels']==1]['english_text'].tolist(),
                            left=False)

(685, 11)
(685, 11)
2545 (2, 2545)
left:  True      feature  pmi_hate  pmi_neutral     score
2280   think -0.558274     0.429985 -0.008656
1029   great -0.640736     0.473054 -0.009756
1671  people -0.640736     0.473054 -0.011382
1333    left -0.892275     0.584085 -0.015087
1357    like -0.558274     0.429985 -0.017313
top_n:              feature  pmi_hate  pmi_neutral     score
985         germany  0.664119    -1.444484  0.043096
655             did  0.596303    -1.164376  0.028274
1977           said  0.721834    -1.749339  0.021645
84         actually  0.721834    -1.749339  0.021645
1266           just  0.285264    -0.389443  0.018714
546         country  0.581657    -1.111909  0.017307
1379           long  0.681192    -1.526946  0.016118
1239        jamaica  0.681192    -1.526946  0.016118
2279         things  0.681192    -1.526946  0.016118
950            free  0.681192    -1.526946  0.016118
1593             oh  0.944227     0.000000  0.012406
1096           help  0.429654    

/nethome/aanegundi/miniconda3/envs/env3/lib/python3.8/site-packages/sklearn/utils/deprecation.py:87: FutureWarning: Function get_feature_names is deprecated; get_feature_names is deprecated in 1.0 and will be removed in 1.2. Please use get_feature_names_out instead.
  warnings.warn(msg, category=FutureWarning)
/nethome/aanegundi/miniconda3/envs/env3/lib/python3.8/site-packages/sklearn/utils/deprecation.py:87: FutureWarning: Function get_feature_names is deprecated; get_feature_names is deprecated in 1.0 and will be removed in 1.2. Please use get_feature_names_out instead.
  warnings.warn(msg, category=FutureWarning)


In [220]:
top_n_left.to_csv('../results/top_n_left_wrong_predictions.csv')
top_n_right.to_csv('../results/top_n_right_wrong_predictions.csv')

In [221]:
pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)
pd.set_option('display.width', 1000)
print(correct_predictions.shape)
correct_predictions = correct_predictions.dropna()
print(correct_predictions.shape)
top_n_left = get_top_n_features_based_pmi(100,
                             correct_predictions[correct_predictions['labels']==0]['english_text'].tolist(),
                             correct_predictions[correct_predictions['labels']==1]['english_text'].tolist(),
                            left=True)
top_n_right = get_top_n_features_based_pmi(100,
                             correct_predictions[correct_predictions['labels']==0]['english_text'].tolist(),
                             correct_predictions[correct_predictions['labels']==1]['english_text'].tolist(),
                            left=False)

(3570, 11)
(3566, 11)
7982 (2, 7982)
left:  True         feature  pmi_hate  pmi_neutral     score
3076     german -1.155279     0.373182 -0.005572
1140  bundestag -0.802608     0.297011 -0.005859
3078    germany -1.205045     0.382238 -0.010238
4620     merkel -0.746960     0.282757 -0.010684
309         afd -2.354563     0.517016 -0.045095
top_n:               feature  pmi_hate  pmi_neutral     score
7679     wagenknecht  1.505774    -6.122528  0.226753
4226            left  1.143596    -1.505868  0.207291
4318           linke  1.492848    -5.162421  0.201562
2069             die  1.398305    -3.016650  0.099046
6645          social  1.218151    -1.789146  0.047226
3262            gysi  1.519320     0.000000  0.045162
3204          gregor  1.519320     0.000000  0.043032
6272           sahra  1.519320     0.000000  0.042606
877            bernd  1.322283    -2.350688  0.042230
6729             spd  0.838780    -0.791994  0.033384
4058         kipping  1.519320     0.000000  0.028120
1

/nethome/aanegundi/miniconda3/envs/env3/lib/python3.8/site-packages/sklearn/utils/deprecation.py:87: FutureWarning: Function get_feature_names is deprecated; get_feature_names is deprecated in 1.0 and will be removed in 1.2. Please use get_feature_names_out instead.
  warnings.warn(msg, category=FutureWarning)
/nethome/aanegundi/miniconda3/envs/env3/lib/python3.8/site-packages/sklearn/utils/deprecation.py:87: FutureWarning: Function get_feature_names is deprecated; get_feature_names is deprecated in 1.0 and will be removed in 1.2. Please use get_feature_names_out instead.
  warnings.warn(msg, category=FutureWarning)


In [222]:
top_n_left.to_csv('../results/top_n_left_correct_predictions.csv')
top_n_right.to_csv('../results/top_n_right_correct_predictions.csv')